### PR TITLE
Decorate reply with nextRenderError

### LINKED ISSUE #435: Unable to render _error page, renders 404 page instead
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.25.0

### Plugin version

7.0.0

### Node.js version

16.13.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.6.1

### Description

The plugin uses `app.render` under the hood, which explicitly disallows rendering `_app`, `_document` and `_error` pages:

https://github.com/vercel/next.js/blob/e75361fd03872b097e817634c049b3185f24cf56/packages/next/server/base-server.ts#L1704

Attempting to render the `_error` path (as described in the readme) results in a 404 Not Found response.

> If you need to render with Next.js from within a custom handler (such as an error handler), use `reply.nextRender`
> 
> ```js
> app.setErrorHandler((err, req, reply) => {
>   reply.status(err.statusCode || 500)
>   return reply.nextRender('/_error')
> })
> ```

### Steps to Reproduce

See description.

### Expected Behavior

The plugin should either:

1. Decorate the request with the `app` instance, so that [`app.renderError`](https://github.com/vercel/next.js/blob/e75361fd03872b097e817634c049b3185f24cf56/packages/next/server/base-server.ts#L2336) can be called manually
2. Decorate fastify with a `reply.nextRenderError` which proxies the `app.renderError` similar to how `reply.nextRender` works

### PR DESCRIPTION
**Pull request by stefee**
Closes #435 

Proxy to Next.js `app.renderError`.

The source code for `app.renderError` at time of writing can be found here:
- [`app.renderError`](https://github.com/vercel/next.js/blob/dda0afd3a45a50cbcb418903f7a0ab56eae51b33/packages/next/server/base-server.ts#L2336) (public)
- [`app.renderErrorToResponse`](https://github.com/vercel/next.js/blob/dda0afd3a45a50cbcb418903f7a0ab56eae51b33/packages/next/server/base-server.ts#L2369) (private)

This method will render an appropriate error page based on the **response status code** - the pathname does not affect the page rendering.

The page that will be rendered by this method can include:

- `/pages/_error.js`
- `/pages/404.js`
- `/pages/500.js`

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -101,6 +102,25 @@ function fastifyNext (fastify, options, next) {
...
+
+    // set custom headers as next will finish the request
+    for (const [headerName, headerValue] of Object.entries(reply.getHeaders())) {
+      // Fastify sets content-length to `undefined` for error handlers, which is an invalid value
+      if (headerName === 'content-length' && headerValue === undefined) {
+        continue
+      }
+
+      reply.raw.setHeader(headerName, headerValue)
+    }
```
Comment (stefee) [inline index.js:118]: copied from `render`  above

File: README.md [hunk 2]
```diff
@@ -49,12 +49,19 @@ fastify.next('/hello', (app, req, reply) => {
...
-  reply.status(err.statusCode || 500)
-  return reply.nextRender('/_error')
+  return reply.nextRender('/a')
+})
+```
+
+If you need to render a Next.js error page, use `reply.nextRenderError`
+
+```js
+app.setErrorHandler((err, req, reply) => {
```
Comment (stefee) [inline README.md:63]: Should this be async ?

```suggestion
app.setErrorHandler(async (err, req, reply) => {
```

Comment (mcollina) [reply README.md:63]: no, it's not needed

**Review by mcollina — COMMENTED** [review]
Can you please add a unit test?

**Comment by stefee** [comment]
@mcollina 
I updated the existing "error page" test to be more accurate - it was not testing anything before other than the ability to call `reply.nextRender` within `fastify.setErrorHandler` - would you prefer I add the existing test back and create a new test?

**Comment by stefee** [comment]
I am unsure if the CI failures are a result of the changes.

**Comment by mcollina** [comment]
It might be, it seems code coverage got reduced.

**Comment by stefee** [comment]
Ah yes, I will look into that.

**Comment by stefee** [comment]
This should be ready for landing :-)

**Pull request merged**